panel/backends/wayland/kwin_wayland/lxqtwmbackend_kwinwayland.cpp

```diff
@@ -0,0 +1,786 @@
+                {
+                    transientsDemandingAttention.insert(leader, window);
+                    emit windowPropertyChanged(leader->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+                }
+            }
+            else if (transientsDemandingAttention.remove(window))
+            {
+                emit windowPropertyChanged(leader->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+            }
+        }
+        else
+        {
+            emit windowPropertyChanged(window->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+        }
+    });
+
+    connect(window, &LXQtTaskBarPlasmaWindow::skipTaskbarChanged, this, [window, this] {
+        updateWindowAcceptance(window);
+    });
+
+    // QObject::connect(window, &PlasmaWindow::applicationMenuChanged, q, [window, this] {
+    //     this->dataChanged(window, QList<int>{ApplicationMenuServiceName, ApplicationMenuObjectPath});
+    // });
+
+    // Handle transient.
+    if (LXQtTaskBarPlasmaWindow *leader = window->parentWindow.data())
+    {
+        transients.insert(window, leader);
+
+        // Update demands attention state for leader.
+        if (window->windowState.testFlag(LXQtTaskBarPlasmaWindow::state::state_demands_attention))
+        {
+            transientsDemandingAttention.insert(leader, window);
+            if(leader->acceptedInTaskBar)
+                emit windowPropertyChanged(leader->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+        }
+    }
+    else
+    {
+        windows.emplace_back(window);
+        updateWindowAcceptance(window);
+    }
+}
+
+bool LXQtWMBackend_KWinWayland::acceptWindow(LXQtTaskBarPlasmaWindow *window) const
+{
+    if(window->windowState.testFlag(LXQtTaskBarPlasmaWindow::state::state_skiptaskbar))
+        return false;
+
+    if(transients.contains(window))
+        return false;
+
+    return true;
+}
+
+void LXQtWMBackend_KWinWayland::updateWindowAcceptance(LXQtTaskBarPlasmaWindow *window)
+{
+    if(!window->acceptedInTaskBar && acceptWindow(window))
+    {
+        window->acceptedInTaskBar = true;
+        emit windowAdded(window->getWindowId());
+    }
+    else if(window->acceptedInTaskBar && !acceptWindow(window))
+    {
+        window->acceptedInTaskBar = false;
+        emit windowRemoved(window->getWindowId());
+    }
+}
+
+LXQtTaskBarPlasmaWindow *LXQtWMBackend_KWinWayland::getWindow(WId windowId) const
+{
+    for(auto &window : std::as_const(windows))
+    {
+        if(window->getWindowId() == windowId)
+            return window.get();
+    }
+
+    return nullptr;
+}
+
+int LXQtWMBackendKWinWaylandLibrary::getBackendScore(const QString &key) const
+{
+    auto *waylandApplication = qGuiApp->nativeInterface<QNativeInterface::QWaylandApplication>();
+    if(!waylandApplication)
+        return 0;
+
+    // Detect KWin Plasma
+    if(key == QLatin1String("KDE") || key == QLatin1String("KWIN"))
```

Comment:
@gfgit We do not have to test for KDE or KWIN, just `kwin_wayland` is sufficient. This is because @stefonarch assures me that `startlxqt-wayland` script sets `XDG_CURRENT_DESKTOP=LXQt:$compositor`. So for kwin, it will be `XDG_CURRENT_DESKTOP=LXQt:kwin_wayland`

Reply:
@gfgit never mind this. I will make this change in my pr.

Reply:
>This is because @stefonarch assures me that startlxqt-wayland script sets XDG_CURRENT_DESKTOP=LXQt:$compositor. So for kwin, it will be XDG_CURRENT_DESKTOP=LXQt:kwin_wayland

Maybe that is fine for kwin, but generally we should cover to use cases: inside LXQt with env vars set by https://github.com/lxqt/lxqt-wayland-session/blob/master/startlxqtwayland.in  and outside, people who just use lxqt-panel in their setup. Probably then they would need to set manually the option in panel.conf I guess.

Reply:
I think we need not worry about it too much. They can surely run lxqt-panel outside of LXQt, but they should be prepared to do a little work. For example, they'll need to run the panel as `XDG_CURRENT_DESKTOP=LXQt:kwin_wayland lxqt-panel`

Reply:
So just the option in `panel.conf` `preferred_backend=xyz` won't do?

Reply:
No, because how do we know for which platform should we load the plugin `xyz`? So the easiest is to specify the platform. `preferred_backend=wayfire:xyz,kwin_wayland:abc`.

Reply:
As of now, it needs a small change, and this won't work. I'll add that change shortly.